### ISSUE TITLE
Unable to Babel

### ISSUE THREAD

**Issue by danilopolani** [jashkenas/underscore#2644]
I can't transform underscore.min with Babel. I need it because, with gulp, I merge all files and then process them with babel, but with underscore it returns an error.

You can try it out here: https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015

At the bottom of the right panel an error appears: "u is undefined".

**Comment by captbaritone** [comment]
I'm getting "Cannot read property '_' of undefined", which is different but also broken.

Which version of Underscore are you using?

I'm suspicious that this is related to https://github.com/jashkenas/underscore/pull/2153 which has been fixed in master, but we have not yet shipped a release that includes it.

You could try it with the version of Underscore on master (https://github.com/jashkenas/underscore/blob/master/underscore.js) and see if that resolves the issue.

Let me know what you find.

**Comment by DaniloPolani** [comment]
I'm using the version 1.8.3, the latest. Unfortunately, I cannot use the unminified version, because of the "require" methods not supported.

> exports is not supported in the browser, you need a commonjs environment such as node.js/io.js, browserify/webpack etc

**Comment by captbaritone** [comment]
Where are you seeing the error with `exports`? I see that message when using the Babel website to transpile it, but the code looks fine. While Underscore references the `exports` variable, it first detects which environment it's in a only uses `exports` if it finds itself running in a node environment.

Basically, that "error" might just be Babel trying too hard to be helpful.

**Comment by rally25rs** [comment]
I'm pretty sure that this happens if Underscore is run through Browserify.
Browserify will turn it into a module, then try running this Underscore code:

```
  // Establish the root object, `window` in the browser, or `exports` on the server.
  var root = this;

  // Save the previous value of the `_` variable.
  var previousUnderscore = root._;
```

But `this` is not the global/window object when it gets there.
Inside a commonJS module, `this` is the same object as `module.exports`.

Browserify is trying to emulate Node's behavior, which is to set `this` to `module.exports` in modules as well. I can't find actual Node docs stating that design decision, but there are several mentions of it on StackOverflow

* [StackOverflow: What does “this” mean in a nodejs module?](http://stackoverflow.com/questions/25058814/what-does-this-mean-in-a-nodejs-module)
* [Mention of it in Browserify GitHub issues (and the issue marked closed because setting this to global is not how Node works)](https://github.com/substack/node-browserify/issues/372)

**Comment by Kosta-Github** [comment]
I think, this is fixed with this commit: https://github.com/jashkenas/underscore/commit/8a847ba69e521b3d7f4faaa6c7c039bcb8e6497b

however, there is no official release available containing this fix; the current version `1.8.3` does not contain it...
